crates/matrix-sdk-base/src/client.rs

```diff
@@ -85,13 +86,13 @@ pub struct BaseClient {
     pub(crate) sync_token: Arc<RwLock<Option<Token>>>,
     /// Database
     store: Store,
+    #[allow(dead_code)]
+    store_path: Option<PathBuf>,
```

Comment:
I think that's a leftover from when we didn't have a `Store`, all the store related stuff and opening your default store should probably be moved into the `Store`.

Not sure if you consider `Store` to be commonly used as well.

Reply:
Oh no, we can't join them, that would introduce a bunch of code and types into the cryptostore and the crypto crate needs to be as lean as possible for the FFI use-case.

`Store` is a wrapper for the specific state store implementation, it provides some simple logic on top and all the additional logic and data to instantiate defaults for the state store could be moved over there.

This would remove `store_path` from the `BaseClient` and make the constructor of the `BaseClient` simpler as well. This would move the questionable feature specific things into a somewhat less commonly used type. But I see you mentioned other ideas in the Matrix room.